src/lambdaconnect_model/scoping.clj

```diff
@@ -328,6 +310,80 @@
 ;                              INTERFACE
 ; ========================================================================================
 
+(defn scope-selected-tags-with-tree
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, map of sets indiacting which tags must be scoped per tag and a set of desired tags.
+   It is advised to calculacte scoping sets once and pass the result.
+  A typical invocation looks like this: 
+  (scope (d/db db/conn) user entities-by-name validated-scope scoping-sets :RARestaurant.ofOwner)))
+
+  Returns a map with db ids, something like:
+  {:RAOwner.me #{11122, 1222} :user #{2312312}}
+  "
+  [config
+   snapshot ; db snapshot
+   user ; user object
+   entities-by-name ; coming from xml model parser
+   scoping-defintion ; the scoping-defintion as read from configuration file
+   tag-scope ; map of sets contating minimal sets of tag which must be evaulated fo given tag (build with get-minim-scoping-set)
+   tags ;set of tags to be scoped
+   ]
+  (let [relevant-rules (->> tags
+                            (map (fn [tag]
+                                   (->> scoping-defintion
+                                        (filter (fn [[k _]] (contains? (tag tag-scope) k)))
+                                        (map (fn [[tag description]] [tag (:constraint description)]))
+                                        (into {}))))
+                            (reduce merge))
+        queries (scoping-step
+                 snapshot
+                 #{(:db/id user)}
+                 entities-by-name
+                 {}
+                 #{:user}
+                 relevant-rules
+                 {:user {:dependencies #{} :rules []}})
+        filtered-queries (into {} (filter (fn [[tag _]] (contains? tags tag)) queries))]
+    (->> filtered-queries
+         (pmap (fn [x] (apply (partial execute-query config) x)))
+         (into {}))))
+
+(defn scope-selected-tags
+  "Takes a snapshot, a user object from DB, entities-by-name, parsed (and validated) EDN of scoping rules, and a set of desired tags.
+   It is advised to calculacte scoping sets once and pass the result.
+  A typical invocation looks like this: 
+  (scope (d/db db/conn) user entities-by-name validated-scope scoping-sets :RARestaurant.ofOwner)))
+
+  Returns a map with db ids, something like:
+  {:RAOwner.me #{11122, 1222} :user #{2312312}}
+  "
+  [config
+   snapshot ; db snapshot
+   user ; user object
+   entities-by-name ; coming from xml model parser
+   scoping-defintion ; the scoping-defintion as read from configuration file 
```

Comment:
Tak a propos spójności - w konwencji Marka ten argument nazywałby się `edn`. Możesz w innych miejscach zmienić to na `scoping-definition` (fajnie jakby wszędzie ten argument nazywałby się tak samo).